scrapy/extensions/logstats_extended.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from datetime import datetime
+
+from twisted.internet import task
+
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.serialize import ScrapyJSONEncoder
+
+logger = logging.getLogger(__name__)
+
+
+class LogStatsExtended:
```

Comment:
I think it may make sense to keep this new class in `log_stats.py`, rather than having a separate file for it.

I also think we may want to reconsider the class name. I would suggest making `Extended` a prefix instead (`ExtendedLogStats`), and consider an alternative word instead (e.g. `Verbose`, `Detailed`, `Flexible`, `Configurable`).

Reply:
I don't have strong opinion about naming anything here.

Except(maybe) to make name.. to avoid disambiguation between this one and default `logstats` extension (assuming that we keep this as separate extension).

In next iteration I will try to use `PERIODIC_LOG_` prefix like`PERIODIC_LOG_STATS` , `PERIODIC_LOG_DELTA` etc.